include/defs/gis.h

```diff
@@ -352,7 +352,8 @@ void G_init_logging(void);
 char *G_file_name(char *, const char *, const char *, const char *);
 char *G_file_name_misc(char *, const char *, const char *, const char *,
 		       const char *);
-char *G_file_name_tmp(char *, const char *, const char *, const char *);
+char *G_file_name_tmp(char *, const char *, const char *, const char *, 
```

Comment:
This change requires an increase of the GRASS major version.
Why not add GRASS_VECTOR_TMPDIR_MAPSET to `G_file_name_tmp()`, update this fn and update relevant code in lib/raster? Most of the other changes in lib/gis could then be removed.